website/content/en/docs/building-operators/golang/quickstart.md

```diff
@@ -37,6 +37,7 @@ Make sure to define `IMG` when you call `make`:
   make docker-build docker-push IMG=$OPERATOR_IMG
   ```
 
+**Note**: You need [kube-builder][kube-builder] installed as the `docker-build` target uses it for the controller test suite.
```

Comment:
Following https://github.com/operator-framework/operator-sdk/pull/4395/files#r556802544: running tests on your project via the makefile should download the test binaries for you. You're saying that isn't happening? You should see something like the following:

```console
$ make docker-build
/go/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."
go fmt ./...
go vet ./...
/go/bin/controller-gen "crd:trivialVersions=true" rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
mkdir -p /go/src/github.com/example/memcached-operator/testbin
test -f /go/src/github.com/example/memcached-operator/testbin/setup-envtest.sh || curl -sSLo /go/src/github.com/example/memcached-operator/testbin/setup-envtest.sh https://raw.githubusercontent.com/kubernetes-sigs/controller-runtime/v0.6.3/hack/setup-envtest.sh
source /go/src/github.com/example/memcached-operator/testbin/setup-envtest.sh; fetch_envtest_tools /go/src/github.com/example/memcached-operator/testbin; setup_envtest_env /go/src/github.com/example/memcached-operator/testbin; go test ./... -coverprofile cover.out
fetching envtest tools@1.16.4 (into '/go/src/github.com/example/memcached-operator/testbin')
kubebuilder/bin/
kubebuilder/bin/etcd
kubebuilder/bin/kube-apiserver
kubebuilder/bin/kubectl
setting up env vars
?   	github.com/example/memcached-operator	[no test files]
?   	github.com/example/memcached-operator/api/v1alpha1	[no test files]
ok  	github.com/example/memcached-operator/controllers	5.210s	coverage: 0.0% of statements
docker build . -t controller:latest
...
```

Reply:
@estroz When I ran `make docker-build IMG=hub.docker.com/$USERNAME/memcached-operator:v0.0.1`, I get this error:

```
[...]
Failure [0.005 seconds]
[BeforeSuite] BeforeSuite 
/home/marty/projects/memcached-operator/controllers/suite_test.go:52

  Unexpected error:
      <*fmt.wrapError | 0xc0003d0ac0>: {
          msg: "failed to start the controlplane. retried 5 times: fork/exec /usr/local/kubebuilder/bin/etcd: no such file or directory",
          err: {
              Op: "fork/exec",
              Path: "/usr/local/kubebuilder/bin/etcd",
              Err: 0x2,
          },
      }
      failed to start the controlplane. retried 5 times: fork/exec /usr/local/kubebuilder/bin/etcd: no such file or directory
  occurred

  /home/marty/projects/memcached-operator/controllers/suite_test.go:62
```

That is when I installed `kube-builder`.

Reply:
Can you post the full output of that command?

Reply:
Don't have the full output as moved on.

Reply:
Hi @hickeyma, 

If you are using the latest version so the error cannot be faced. See that the `docker-build` will call the test target which will install the envtest binaries if required. Then, IMO it should not be added at all. If you are facing problems with that then it would mean that is a bug. 

Users should not care at all to install kube-builder. 

/hold

Reply:
Ok, retraced my steps and realised there was earlier errors  as follows:

```console
$ make docker-build IMG=docker.io/$USERNAME/memcached-operator:v0.0.1
/home/marty/projects/memcached-operator/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."
go fmt ./...
go vet ./...
/home/marty/projects/memcached-operator/bin/controller-gen "crd:trivialVersions=true,preserveUnknownFields=false" rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
mkdir -p /home/marty/projects/memcached-operator/testbin
test -f /home/marty/projects/memcached-operator/testbin/setup-envtest.sh || curl -sSLo /home/marty/projects/memcached-operator/testbin/setup-envtest.sh https://raw.githubusercontent.com/kubernetes-sigs/controller-runtime/v0.7.0/hack/setup-envtest.sh
source /home/marty/projects/memcached-operator/testbin/setup-envtest.sh; fetch_envtest_tools /home/marty/projects/memcached-operator/testbin; setup_envtest_env /home/marty/projects/memcached-operator/testbin; go test ./... -coverprofile cover.out
/bin/sh: 1: source: not found
/bin/sh: 1: fetch_envtest_tools: not found
/bin/sh: 1: setup_envtest_env: not found
?   	github.com/example/memcached-operator	[no test files]
?   	github.com/example/memcached-operator/api/v1alpha1	[no test files]
Running Suite: Controller Suite
===============================
[...]
```
This led to the binaries for test not been downloaded and the further on error in the test suite (in https://github.com/operator-framework/operator-sdk/pull/4395#discussion_r559597657). I traced it down to the use of `source` in https://github.com/operator-framework/operator-sdk/blob/master/testdata/go/v2/memcached-operator/Makefile#L33. I am using Ubuntu and `sh` points to `dash` by default and as `source` is a bash command, it cannot be found. Hence, it fails when trying to reference `source` and requires specifying the `bash` shell in the `Makefile`.

I then came across issue #4203 which covers the issue. It is good that work has been done on this by @sayboras and @estroz; the fix pushed in https://github.com/kubernetes-sigs/controller-runtime/pull/1258 seems like a reasonable approach. I am just wondering then if it is worth adding a note in the meantime (until the fix becomes available) along the lines of:

```
Note: If using an OS which does not point `sh` to the `bash` shell (Ubuntu for example) then you should add the following line to the `Makefile`:
SHELL := /bin/bash

This will fix potential issue running the the controller test suite, similar to:
failed to start the controlplane. retried 5 times: fork/exec /usr/local/kubebuilder/bin/etcd: no such file or directory
  occurred
```

Reply:
Sounds good to me, thanks!

Reply:
Updated the note to specify the bash shell if needed. Removed about `kubebuilder`.